core/server/data/schema/commands.js

```diff
@@ -258,64 +231,67 @@ async function dropForeign({fromTable, fromColumn, toTable, toColumn, transactio
         }
-    } else {
-        logging.warn(`Skipped dropping foreign key from ${fromTable}.${fromColumn} to ${toTable}.${toColumn} - foreign key does not exist`);
+    } catch (err) {
+        if (err.code === 'ER_CANT_DROP_FIELD_OR_KEY') {
+            logging.warn(`Skipped dropping foreign key from ${fromTable}.${fromColumn} to ${toTable}.${toColumn} - foreign key does not exist`);
+            return;
+        }
+        throw err;
     }
 }
 
 /**
  * Checks if primary key index exists in a table over the given columns.
  *
- * @param {string} tableName - name of the table to add primary key constraint to
- * @param {Object} transaction - connnection object containing knex reference
- * @param {Object} transaction.knex - knex instance
+ * @param {string} tableName - name of the table to check primary key constraint on
+ * @param {import('knex')} transaction - connnection object containing knex reference
  */
-async function hasPrimaryKey(tableName, transaction) {
+async function hasPrimaryKeySQLite(tableName, transaction) {
     const knex = (transaction || db.knex);
     const client = knex.client.config.client;
 
-    if (client === 'mysql') {
-        const dbName = knex.client.config.connection.database;
-        const [rawConstraints] = await knex.raw(`
-                SELECT CONSTRAINT_NAME
-                FROM information_schema.TABLE_CONSTRAINTS
-                WHERE 1=1
-                AND CONSTRAINT_SCHEMA=:dbName
-                AND TABLE_NAME=:tableName
-                AND CONSTRAINT_TYPE='PRIMARY KEY'`, {dbName, tableName});
-
-        return rawConstraints.length > 0;
-    } else {
-        const rawConstraints = await knex.raw(`PRAGMA index_list('${tableName}');`);
-        const tablePrimaryKey = rawConstraints.find(c => c.origin === 'pk');
-
-        return tablePrimaryKey;
+    if (client !== 'sqlite3') {
+        throw new Error('Must use hasPrimaryKeySQLite on an SQLite3 database');
     }
+
+    const rawConstraints = await knex.raw(`PRAGMA index_list('${tableName}');`);
+    const tablePrimaryKey = rawConstraints.find(c => c.origin === 'pk');
+
+    return tablePrimaryKey;
 }
 
 /**
  * Adds an primary key index to a table over the given columns.
  *
  * @param {string} tableName - name of the table to add primaykey  constraint to
  * @param {string|[string]} columns - column(s) to form primary key constraint with
- * @param {Object} transaction - connnection object containing knex reference
- * @param {Object} transaction.knex - knex instance
+ * @param {import('knex')} transaction - connnection object containing knex reference
  */
 async function addPrimaryKey(tableName, columns, transaction) {
-    const hasUniqueConstraint = await hasPrimaryKey(tableName, transaction);
-
-    if (!hasUniqueConstraint) {
+    const isSQLite = db.knex.client.config.client === 'sqlite3';
+    if (isSQLite) {
+        const primaryKeyExists = await hasPrimaryKeySQLite(tableName, transaction);
+        if (!primaryKeyExists) {
```

Comment:
@allouis Should this be:

```suggestion
        if (primaryKeyExists) {
```